### PR TITLE
Accounts V2: Refactor Wallet Open / Close and Keymanager Initialization

### PR DESCRIPTION
**Pull request by rauljordan**
**What type of PR is this?**

> Feature

**What does this PR do? Why is it needed?**

This PR refactors the accounts-v2 code a bit to allow for simpler user experience when creating a derived account. The only required user input is the wallet password, which will be used to unlock and decrypt the seed phrase which can then be used to generate as many accounts as the user pleases. Additionally, this PR includes the following refactors:

1. Simplify the `Keymanager` interface to be more straightforward. All it contains is `FetchValidatingPublicKeys` and `Sign`
2. Utilize a switch statement to decide how to create an account depending on the kind of keymanager a wallet uses instead of relying on one-size-fits-all interface
3. Make the `OpenWallet` function smarter, being able to detect the type of keymanager from disk as well as asking for a wallet password + passwords directory if relevant. This reduced the need for boilerplate and made it simple to open a wallet.
4. Adds implicit handling of wallet dir if a wallet exists at the default path

**Which issues(s) does this PR fix?**

Part of #6220

### CODE REVIEW

File: validator/accounts/v2/new.go [hunk 1]
```diff
@@ -42,43 +44,44 @@ func NewAccount(cliCtx *cli.Context) error {
...
-
-	// Only direct keymanagers can create accounts for now.
-	if keymanagerKind == v2keymanager.Remote {
-		log.Fatal("Cannot create a new account for a remote keymanager")
-	}
-	// Read the directory for password storage from user input.
-	passwordsDirPath := inputPasswordsDirectory(cliCtx)
-	wallet, err := OpenWallet(ctx, &WalletConfig{
-		PasswordsDir:      passwordsDirPath,
+	wallet, err := OpenWallet(cliCtx, &WalletConfig{
```
Comment (rauljordan) [inline validator/accounts/v2/new.go:47]: Simplified a lot of logic by making OpenWallet smarter, being able to detect the type of keymanager, ask for a wallet password if none specified, and more

File: validator/accounts/v2/wallet.go [hunk 2]
```diff
@@ -102,18 +102,29 @@ func NewWallet(ctx context.Context, cfg *WalletConfig) (*Wallet, error) {
 // OpenWallet instantiates a wallet from a specified path. It checks the
 // type of keymanager associated with the wallet by reading files in the wallet
 // path, if applicable. If a wallet does not exist, returns an appropriate error.
-func OpenWallet(ctx context.Context, cfg *WalletConfig) (*Wallet, error) {
+func OpenWallet(cliCtx *cli.Context, cfg *WalletConfig) (*Wallet, error) {
```
Comment (rauljordan) [inline validator/accounts/v2/wallet.go:105]: This function is now smarter, reducing the need for boilerplate. It is able to determine the kind of keymanager being used as well as the passwords directory if using a direct keymanager

File: validator/accounts/v2/list.go [hunk 3]
```diff
@@ -19,39 +19,30 @@ import (
 
 // ListAccounts displays all available validator accounts in a Prysm wallet.
 func ListAccounts(cliCtx *cli.Context) error {
-	walletDir, err := inputWalletDir(cliCtx)
```
Comment (rauljordan) [inline validator/accounts/v2/list.go:22]: No more of this boilerplate, open wallet now does this internally

File: validator/accounts/v2/list.go [hunk 4]
```diff
@@ -84,8 +75,6 @@ func listDirectKeymanagerAccounts(
 		au.BrightRed("View the eth1 deposit transaction data for your accounts " +
 			"by running `validator accounts-v2 list --show-deposit-data"),
 	)
-	dirPath := au.BrightCyan("(wallet dir)")
-	fmt.Printf("%s %s\n", dirPath, wallet.AccountsDir())
```
Comment (0xKiwi) [inline validator/accounts/v2/list.go:88]: Why was this removed?
